File: 2/2.js
var fib = [1,2];

while (fib[fib.length-1]+fib[fib.length-2] <= 4000000) {

	fib.push(fib[fib.length-1]+fib[fib.length-2]);

}



console.log(fib.filter(function(number){
	return number % 2 == 0;
}).reduce(function(a,b){
	return a+b;
},0));

File: 2/2.py
fib = [1,2]

while fib[-1]+fib[-2] < 4000000:
	fib.append(fib[-1]+fib[-2])

evens = filter(lambda f: f % 2 == 0,fib)

print reduce(lambda a, b: a+b, evens)